### PR TITLE
Moving task_group_exptensions out of the preview

### PR DESCRIPTION
**Pull request by anton-potapov**
### Description 
- Moved `task_handle` and related out of  Preview features to be aligned with [
oneAPI V1.1 rev 1](https://github.com/oneapi-src/oneAPI-spec/releases/tag/oneapi-v1.1-rev-1) :

Following features, as not part of the spec are still preview ones:
  - `is_inside_task` function
  - scheduler bypass via `task_handle`

Fixes # - _issue number(s) if exists_

- [x] - git commit message contains appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/oneapi-src/oneTBB/blob/master/CONTRIBUTING.md#pull-requests) for details)_

### Type of change

_Choose one or multiple, leave empty if none of the other choices apply_

_Add respective label(s) to PR if you have permissions_

- [ ] bug fix - _change which fixes an issue_
- [ ] new feature - _change which adds functionality_
- [ ] tests - _change in tests_
- [ ] infrastructure - _change in infrastructure and CI_
- [ ] documentation - _documentation update_

### Tests

- [x] added - _required for new features and for some bug fixes_
- [ ] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [x] needs to be updated
- [ ] not needed

### Breaks backward compatibility
- [ ] Yes
- [x] No
- [ ] Unknown

### Notify the following users
_List users with `@` to send notifications_

### CODE REVIEW

File: test/conformance/conformance_task_group.cpp [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+        oneapi::tbb::task_group tg;
+
+        std::atomic<bool> run {false};
+
+        auto h = tg.defer([&]{
+            run = true;
+        });
+        CHECK_MESSAGE(run == false, "delayed task should not be run until run(task_handle) is called");
+    }
+}
```
Comment (isaevil) [inline test/conformance/conformance_task_group.cpp:56]: Duplicates test case on line 30.

Comment (anton-potapov) [reply test/conformance/conformance_task_group.cpp:40]: fixed

**Comment by isaevil** [comment]
Should we check somewhere for `task_handle` being only movable?

**Comment by anton-potapov** [comment]
> 

added `static_assert`ion test

**Comment by alongwithyou** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

File: test/conformance/conformance_task_group.cpp [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        tg.wait();
+        CHECK_MESSAGE(completed == true, "Deferred task should be completed when task_group::wait exits");
+    }};
+
+    utils::SpinWaitUntilEq(thread_started, true);
+    CHECK_MESSAGE(completed == false, "Deferred task should not be run until run(task_handle) is called");
+
+    tg.run(std::move(h));
+    //TODO: more accurate test (with fixed number of threads (1 ?) to guarantee correctness of following assert)
+    //CHECK_MESSAGE(completed == false, "Deferred task should not be run until run(task_handle) and wait is called");
```
Comment (alexey-katranov) [inline test/conformance/conformance_task_group.cpp:150]: I am not sure that I understand the TODO. Why deferred task should not be run until wait is called?

File: test/conformance/conformance_task_group.cpp [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+              (!std::is_copy_constructible<oneapi::tbb::task_handle>::value)
+            &&(!std::is_copy_assignable<oneapi::tbb::task_handle>::value),
+            "oneapi::tbb::task_handle should be non copyable");
+}
+//! Test that task_handle prolongs task_group::wait
+//! \brief \ref requirement
+TEST_CASE("Task handle blocks wait"){
+    oneapi::tbb::task_group tg;
+
+    std::atomic<bool> completed  {false};
```
Comment (alexey-katranov) [inline test/conformance/conformance_task_group.cpp:128]: Make `completed` to be non-atomic

File: test/conformance/conformance_task_group.cpp [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+        });
+
+        utils::try_call([&] {
+            std::forward<CancelF>(cancel)(outer_tg);
+        }).on_completion([&] {
+            outer_cancelled = true;
+        });
+    };
+
+    auto check = [&] {
```
Comment (alexey-katranov) [inline test/conformance/conformance_task_group.cpp:209]: Why do we need a lambda for check?

File: test/conformance/conformance_task_group_context.cpp [hunk 5]
```diff
@@ -52,3 +52,5 @@ TEST_CASE("Test methods") {
     CHECK_FALSE(ctx.is_group_execution_cancelled());
     ctx.traits();
 }
+
+//todo: add test for task_group_context(task_group_context*)
```
Comment (alexey-katranov) [inline test/conformance/conformance_task_group_context.cpp:56]:
```suggestion
//TODO: add test for task_group_context(task_group_context*)
```

File: test/conformance/conformance_task_arena.cpp [hunk 6]
```diff
@@ -118,3 +118,103 @@ TEST_CASE("Task arena copy constructor") {
...
+    oneapi::tbb::task_group tg;
+
+    std::atomic<bool> run{false};
+
+    auto task_handle = tg.defer([&]{ run = true; });
+
+    arena.enqueue(std::move(task_handle));
+    tg.wait();
+
+    CHECK(run == true);
```
[the remark this answers is not in the document]
Comment (anton-potapov) [reply test/conformance/conformance_task_arena.cpp]: fixed

File: test/conformance/conformance_task_arena.cpp [hunk 7]
```diff
@@ -118,3 +118,103 @@ TEST_CASE("Task arena copy constructor") {
...
+}
+
+//! Basic test for this_task_arena::enqueue with task handle
+//! \brief \ref interface \ref requirement
+TEST_CASE("this_task_arena::enqueue task_handle") {
+    oneapi::tbb::task_arena arena;
+    oneapi::tbb::task_group tg;
+
+    std::atomic<bool> run{false};
```
[the remark this answers is not in the document]
Comment (anton-potapov) [reply test/conformance/conformance_task_arena.cpp]: fixed

File: test/conformance/conformance_task_arena.cpp [hunk 8]
```diff
@@ -118,3 +118,103 @@ TEST_CASE("Task arena copy constructor") {
...
+//TODO: Add
+//! Basic test for this_task_arena::enqueue with functor
+
+//! Test case for the common use-case of prolonging task_group lifetime
+//! \brief \ref interface \ref requirement
+TEST_CASE("this_task_arena::enqueue prolonging task_group") {
+    oneapi::tbb::task_arena arena;
+    oneapi::tbb::task_group tg;
+
+    std::atomic<bool> run{false};
```
[the remark this answers is not in the document]
Comment (anton-potapov) [reply test/conformance/conformance_task_arena.cpp]: fixed

File: test/conformance/conformance_task_group.cpp [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+TEST_CASE("Task handle created but not run"){
+    {
+        oneapi::tbb::task_group tg;
+
+        std::atomic<bool> run {false};
+
+        auto h = tg.defer([&]{
+            run = true;
+        });
+        CHECK_MESSAGE(run == false, "delayed task should not be run until run(task_handle) is called");
```
[the remark this answers is not in the document]
Comment (anton-potapov) [reply test/conformance/conformance_task_group.cpp]: fixed

File: test/conformance/conformance_task_group.cpp [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+    }
+}
+
+//! Basic test for task handle
+//! \brief \ref interface \ref requirement
+TEST_CASE("Task handle run"){
+    oneapi::tbb::task_handle h;
+
+    oneapi::tbb::task_group tg;
+    std::atomic<bool> run {false};
```
[the remark this answers is not in the document]
Comment (anton-potapov) [reply test/conformance/conformance_task_group.cpp]: fixed

File: test/conformance/conformance_task_group.cpp (hunk 3 above)
Comment (anton-potapov) [reply test/conformance/conformance_task_group.cpp]: fixed

File: test/conformance/conformance_task_group.cpp (hunk 2 above)
Comment (anton-potapov) [reply test/conformance/conformance_task_group.cpp:158]: IIRC, the idea was to make a test more stable by making it single-threaded and utilize the fact that task added by `run` is not guaranteed to execute until `wait` is called. 

BTW for some unknown reason [task_group spec](https://spec.oneapi.io/versions/latest/elements/oneTBB/source/task_scheduler/task_group/task_group_cls.html) does not mention this.

File: test/conformance/conformance_task_group.cpp (hunk 4 above)
Comment (anton-potapov) [reply test/conformance/conformance_task_group.cpp:217]: I have no idea :) maybe @aleksei-fedotov has

Comment (aleksei-fedotov) [reply test/conformance/conformance_task_group.cpp:217]: Obviously, the lambda is not needed to be here and the encapsulated code can be unwrapped.

**Review by alexey-katranov — APPROVED** [review]

**Pull request merged**